idseq_dag/engine/pipeline_flow.py

```diff
@@ -126,6 +126,18 @@ def plan(self):
             covered_nodes.update(current_nodes)
         return (step_list, large_file_download_list, covered_nodes)
 
+    @staticmethod
+    def fetch_input_files_from_s3(input_files, input_dir_s3, result_dir_local):
+        for f in input_files:
+            s3_file = os.path.join(input_dir_s3, f)
+            local_file = os.path.join(result_dir_local, f)
+            # copy the file over
+            subprocess.check_call("aws s3 cp %s %s/" % (s3_file, result_dir_local), shell=True)
+
+            # write the done_file
+            done_file = PipelineStep.done_file(local_file)
+            subprocess.check_call("date > %s" % done_file, shell=True)
```

Comment:
Is this done file supposed to be written always, or only when the step succeeded?

Reply:
Obviously this shouldn't be sequential, but I guess that will be my job? :)

Reply:
yes. when the step succeeded but in this case it's downloaded from s3 so we need to do this.